### ISSUE TITLE
Deployment never completes but cluster is active

### ISSUE THREAD

**Issue by Tomthi** [terraform-aws-modules/terraform-aws-eks#777]
# I have issues
The same code was use to build two other clusters and they are working. There was some lag from when the initial clusters were built. 

## I'm submitting a...

* [X] bug report
* [ ] feature request
* [ ] support request - read the [FAQ](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/docs/faq.md) first!
* [ ] kudos, thank you, warm fuzzy

## What is the current behavior?

The cluster deploy never completes with worker nodes and ASG's.  The Cluster state does show active. Seems like the ASG' and worker nodes are never deployed. 

## If this is a bug, how to reproduce? Please include a code sample if relevant.

When running apply, the cluster is built and in an active state but the node groups are never created, there are no ASG's or launch templates. 

It's stuck in a loop
module.eks-cluster.null_resource.wait_for_cluster[0]: Still creating...
module.eks-cluster.null_resource.wait_for_cluster[0]: Still creating...
module.eks-cluster.null_resource.wait_for_cluster[0]: Still creating...
module.eks-cluster.null_resource.wait_for_cluster[0]: Still creating...

Main.tf
module "eks-cluster" {
  source                          = "terraform-aws-modules/eks/aws"
  version                         = "9.0.0"
  cluster_name                   = "qa-eps-eks"
 subnets                             =  "${data.aws_subnet_ids.default.ids}"
  vpc_id                              =  "${var.vpc_id}"
  cluster_endpoint_private_access = true
  cluster_endpoint_public_access  = false
  cluster_security_group_id =  "${data.aws_security_group.default.id}"

  worker_groups = [
    {
      name = "qa-eks-workers"
      instance_type = "m4.large"
      key_name = "qa"
      asg_min_size      = 2
      asg_desired_size  = 2
      asg_max_size      = 5
      autoscaling_enabled = true
      tags = [{
        propagate_at_launch = true
        key         = "terraform"
        value       = "true"
      }]
    }
  ]
  tags = {
    environment = "qa"
    terraform   = "true"
  }
}

## What's the expected behavior?

The expected behavior is the cluster is built along with ASG and worker nodes.

## Environment details

* Affected module version:
* OS:
* Terraform version: 0.12.23

## Any other relevant info 

Module v9.0.0

**Comment by js-timbirkett** [comment]
Do you have wget installed? The command that gets run is: https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/variables.tf#L204

I'm not sure how the module would behave in terms of redirecting  stderr but testing the command with a non existing command on my machine:
```
❯ until wgett --no-check-certificate -O - -q $ENDPOINT/healthz >/dev/null; do echo "sleeping" ; sleep 4; done
zsh: command not found: wgett
sleeping
zsh: command not found: wgett
sleeping
zsh: command not found: wgett
sleeping
zsh: command not found: wgett
sleeping
```

it runs forever as far as I can see.

**Comment by daroga0002** [comment]
If wget will be not available then it will fail. In this case I suggest to check wget version as there is known issue with SSL compatibility or verify does endpoint is really available from place where you running terraform code (in case of private envpoint it must be same VPC, in case of public endpoint you must ensure about whitelist value configured under https://github.com/terraform-aws-modules/terraform-aws-eks/blob/4c0c4c49a611e0d7dcaff334fc7862c31732d1f0/variables.tf#L249-L253 ) 

Please check  also https://github.com/terraform-aws-modules/terraform-aws-eks/issues/757

**Comment by js-timbirkett** [comment]
I see the problem I think:

```
cluster_endpoint_private_access = true
cluster_endpoint_public_access = false
```

**Comment by Tomthi** [comment]
@js-timbirkett That seems to have fixed it. I reverted to v8.1.0 and then went back to v9.0.0 removed the one line

cluster_endpoint_public_access = false

It also didn't like, so I had to fix that. 
asg_min_size = 2
asg_desired_size = 2
asg_max_size = 5

So everything looks like it's provisioned, but it does throw an error about the auth map already exists. I did clear out the authmap that it drops in the root and deleted .terraform it it still throws that error at the end.
